docs/extensions/constraints.md

```diff
@@ -0,0 +1,131 @@
+# Supporting Global Constraints in gittuf
+
+Last Modified: January 16, 2025
+
+Status: Draft
+
+gittuf currently supports explicitly granting write permission to a namespace to
+a specific set of actors as well as a threshold that identifies the number of
+actors that must agree to a change to that namespace. This extension of trust
+uses the concept of "delegation", and a set of actors granted some trust can
+choose to delegate this further to other actors.
+
+This mechanism does not support setting generic controls over changes in the
+repository, perhaps to enforce a baseline security posture _irrespective of the
+specific delegations controlling who can make changes to a namespace_. For
+example, a repository's owners may choose to block force pushes to some
+branches, thus preventing the rewriting of history for that branch.
+Alternatively, a repository's owners may require all changes to some branch to
+receive at least two approvals, irrespective of the specific actors trusted,
+which can be declared through multiple levels of delegations.
+
+## Declaring Global Constraints
+
+This extension introduces the notion of "global constraints" (also referred to
+as "global rules") that are declared by the repository's owners in the root of
+trust metadata. Unlike the delegations implemented in gittuf today, global
+constraints can be of distinct types, with each type indicating how it is
+verified by a gittuf client. As with any changes to root of trust metadata,
+adding, removing, or updating a global constraint requires approval from a
+threshold of repository owners.
```

Comment:
Owners for which repository?

Reply:
This document only considers the context of a single repository. While it's currently the same PR, this document doesn't relate to gittuf across repositories.